src/super_gradients/training/models/detection_models/ssd.py

```diff
@@ -148,11 +153,29 @@ def forward(self, x):
             x = block(x)
             detection_feed.append(x)
 
-        # FEATURE MAPS: i.e. FOR 300X300 INPUT - 38X38X4, 19X19X6, 10X10X6, 5X5X6, 3X3X4, 1X1X4
+        # detection_feed are FEATURE MAPS: i.e. FOR 300X300 INPUT - 38X38X4, 19X19X6, 10X10X6, 5X5X6, 3X3X4, 1X1X4
         locs, confs = self.bbox_view(detection_feed)
 
-        # FOR 300X300 INPUT - RETURN N_BATCH X 8732 X {N_LABELS, N_LOCS} RESULTS
-        return locs, confs
+        if self.training:
+            # FOR 300X300 INPUT - RETURN N_BATCH X 8732 X {N_LABELS, N_LOCS} RESULTS
+            return locs, confs
+        else:
```

Comment:
This means that we aren't able to visualise training predictions then...If this takes a lot of time we can just drop it, but otherwise maybe we should reconsider.